### PR TITLE
GetActivationEpoch implementation + further refactor

### PR DESCRIPTION
**Pull request by sstanculeanu**
## Reasoning behind the pull request
- further integration/implementation
  
## Proposed changes
- added implementation for GetActivaitonEpoch and integrated it

## Testing procedure
- will be tested with feat branch

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: common/constants.go [hunk 1]
```diff
@@ -987,5 +987,8 @@ const (
 	DeterministicSortOnValidatorsInfoFixFlag           core.EnableEpochFlag = "DeterministicSortOnValidatorsInfoFixFlag"
 	DynamicGasCostForDataTrieStorageLoadFlag           core.EnableEpochFlag = "DynamicGasCostForDataTrieStorageLoadFlag"
 	ScToScLogEventFlag                                 core.EnableEpochFlag = "ScToScLogEventFlag"
+  BlockGasAndFeesReCheckFlag                         core.EnableEpochFlag = "BlockGasAndFeesReCheckFlag"
```
Comment (BeniaminDrasovean) [inline common/constants.go:990]: This is not alligned

File: common/enablers/enableEpochsHandler.go [hunk 2]
```diff
@@ -42,312 +47,636 @@ func NewEnableEpochsHandler(enableEpochsConfig config.EnableEpochs, epochNotifie
...
+		},
+		common.ESDTMetadataContinuousCleanupFlag: {
+			isActiveInEpoch: func(epoch uint32) bool {
+				return epoch >= handler.enableEpochsConfig.ESDTMetadataContinuousCleanupEnableEpoch
+			},
+			activationEpoch: handler.enableEpochsConfig.ESDTMetadataContinuousCleanupEnableEpoch,
+		},
+		common.FixAsyncCallbackCheckFlag: {
+			isActiveInEpoch: func(epoch uint32) bool {
+				return epoch >= handler.enableEpochsConfig.ESDTMetadataContinuousCleanupEnableEpoch
```
Comment (AdoAdoAdo) [inline common/enablers/enableEpochsHandler.go:491]: is the flag correct?

File: common/enablers/enableEpochsHandler.go [hunk 3]
```diff
@@ -42,312 +47,636 @@ func NewEnableEpochsHandler(enableEpochsConfig config.EnableEpochs, epochNotifie
...
+			},
+			activationEpoch: handler.enableEpochsConfig.ESDTMetadataContinuousCleanupEnableEpoch,
+		},
+		common.SendAlwaysFlag: {
+			isActiveInEpoch: func(epoch uint32) bool {
+				return epoch >= handler.enableEpochsConfig.ESDTMetadataContinuousCleanupEnableEpoch
```
Comment (AdoAdoAdo) [inline common/enablers/enableEpochsHandler.go:497]: is the flag correct?

File: common/enablers/enableEpochsHandler.go [hunk 4]
```diff
@@ -42,312 +47,636 @@ func NewEnableEpochsHandler(enableEpochsConfig config.EnableEpochs, epochNotifie
...
+			},
+			activationEpoch: handler.enableEpochsConfig.ESDTMetadataContinuousCleanupEnableEpoch,
+		},
+		common.ChangeDelegationOwnerFlag: {
+			isActiveInEpoch: func(epoch uint32) bool {
+				return epoch >= handler.enableEpochsConfig.ESDTMetadataContinuousCleanupEnableEpoch
```
Comment (AdoAdoAdo) [inline common/enablers/enableEpochsHandler.go:503]: is the flag correct?

File: common/constants.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply common/constants.go:990]: nope, fixed

File: common/enablers/enableEpochsHandler.go [hunk 5]
```diff
@@ -42,312 +47,636 @@ func NewEnableEpochsHandler(enableEpochsConfig config.EnableEpochs, epochNotifie
...
+		},
+		common.RemoveNonUpdatedStorageFlag: {
+			isActiveInEpoch: func(epoch uint32) bool {
+				return epoch >= handler.enableEpochsConfig.RemoveNonUpdatedStorageEnableEpoch
+			},
+			activationEpoch: handler.enableEpochsConfig.RemoveNonUpdatedStorageEnableEpoch,
+		},
+		common.OptimizeNFTStoreFlag: {
+			isActiveInEpoch: func(epoch uint32) bool {
+				return epoch >= handler.enableEpochsConfig.OptimizeNFTStoreEnableEpoch
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply common/enablers/enableEpochsHandler.go:359]: yes, [rc/v1.6.0 proof](https://github.com/multiversx/mx-chain-go/blob/rc/v1.6.0/common/enablers/enableEpochsHandler.go#L89)

File: common/enablers/enableEpochsHandler.go [hunk 6]
```diff
@@ -42,312 +47,636 @@ func NewEnableEpochsHandler(enableEpochsConfig config.EnableEpochs, epochNotifie
...
+			},
+			activationEpoch: handler.enableEpochsConfig.OptimizeNFTStoreEnableEpoch,
+		},
+		common.SaveToSystemAccountFlag: {
+			isActiveInEpoch: func(epoch uint32) bool {
+				return epoch >= handler.enableEpochsConfig.OptimizeNFTStoreEnableEpoch
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply common/enablers/enableEpochsHandler.go:365]: yes, [rc/v1.6.0 proof](https://github.com/multiversx/mx-chain-go/blob/rc/v1.6.0/common/enablers/epochFlags.go#L600)

File: common/enablers/enableEpochsHandler.go [hunk 7]
```diff
@@ -42,312 +47,636 @@ func NewEnableEpochsHandler(enableEpochsConfig config.EnableEpochs, epochNotifie
...
+			},
+			activationEpoch: handler.enableEpochsConfig.OptimizeNFTStoreEnableEpoch,
+		},
+		common.CheckFrozenCollectionFlag: {
+			isActiveInEpoch: func(epoch uint32) bool {
+				return epoch >= handler.enableEpochsConfig.OptimizeNFTStoreEnableEpoch
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply common/enablers/enableEpochsHandler.go:371]: yes, [rc/v1.6.0 proof](https://github.com/multiversx/mx-chain-go/blob/rc/v1.6.0/common/enablers/epochFlags.go#L606)

File: common/enablers/enableEpochsHandler.go [hunk 8]
```diff
@@ -42,312 +47,636 @@ func NewEnableEpochsHandler(enableEpochsConfig config.EnableEpochs, epochNotifie
...
+			},
+			activationEpoch: handler.enableEpochsConfig.OptimizeNFTStoreEnableEpoch,
+		},
+		common.ValueLengthCheckFlag: {
+			isActiveInEpoch: func(epoch uint32) bool {
+				return epoch >= handler.enableEpochsConfig.OptimizeNFTStoreEnableEpoch
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply common/enablers/enableEpochsHandler.go:377]: yes, [rc/v1.6.0 proof](https://github.com/multiversx/mx-chain-go/blob/rc/v1.6.0/common/enablers/epochFlags.go#L618)

File: common/enablers/enableEpochsHandler.go [hunk 9]
```diff
@@ -42,312 +47,636 @@ func NewEnableEpochsHandler(enableEpochsConfig config.EnableEpochs, epochNotifie
...
+			},
+			activationEpoch: handler.enableEpochsConfig.OptimizeNFTStoreEnableEpoch,
+		},
+		common.CheckTransferFlag: {
+			isActiveInEpoch: func(epoch uint32) bool {
+				return epoch >= handler.enableEpochsConfig.OptimizeNFTStoreEnableEpoch
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply common/enablers/enableEpochsHandler.go:383]: yes, [rc/v1.6.0 proof](https://github.com/multiversx/mx-chain-go/blob/rc/v1.6.0/common/enablers/epochFlags.go#L624)

File: common/enablers/enableEpochsHandler.go (hunk 2 above)
Comment (sstanculeanu) [reply common/enablers/enableEpochsHandler.go:491]: yes, [rc/v1.6.0 proof](https://github.com/multiversx/mx-chain-go/blob/rc/v1.6.0/common/enablers/epochFlags.go#L594)

File: common/enablers/enableEpochsHandler.go (hunk 3 above)
Comment (sstanculeanu) [reply common/enablers/enableEpochsHandler.go:497]: yes, [rc/v1.6.0 proof](https://github.com/multiversx/mx-chain-go/blob/rc/v1.6.0/common/enablers/epochFlags.go#L612)

File: common/enablers/enableEpochsHandler.go (hunk 4 above)
Comment (sstanculeanu) [reply common/enablers/enableEpochsHandler.go:503]: yes, [rc/v1.6.0 proof](https://github.com/multiversx/mx-chain-go/blob/rc/v1.6.0/common/enablers/enableEpochsHandler.go#L112)

**Pull request merged**